src/Sarif.Multitool/Rules/SARIF2001.AuthorHighQualityMessages.cs

```diff
@@ -3,85 +3,106 @@
 
 using System;
 using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
 using Microsoft.Json.Pointer;
 
 namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
 {
     public class AuthorHighQualityMessages : SarifValidationSkimmerBase
     {
-        public override MultiformatMessageString FullDescription => new MultiformatMessageString
-        {
-            Text = RuleResources.SARIF2001_AuthorHighQualityMessages_FullDescription_Text
+        /// <summary>
+        /// SARIF2001
+        /// </summary>
+        public override string Id => RuleId.AuthorHighQualityMessages;
+
+        /// <summary>
+        /// Placeholder (full description).
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF2001_AuthorHighQualityMessages_FullDescription_Text };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_EnquoteDynamicContent_Text),
+            nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_IncludeDynamicContent_Text),
+            nameof(RuleResources.SARIF2001_AuthorHighQualityMessages_Warning_TerminateWithPeriod_Text)
         };
 
         public override FailureLevel DefaultLevel => FailureLevel.Warning;
 
-        public override string Id => RuleId.AuthorHighQualityMessages;
+        private static readonly Regex s_dynamicContentRegex = new Regex(@"{[0-9]+}", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+        private static readonly Regex s_enquoteDynamicContentRegex = new Regex(@"'{[0-9]+}'", RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

Comment:
>s_enquoteDynamicContentRegex [](start = 38, length = 28)

How about this:

```
private static readonly Regex s_nonEnquotedDynamicContextRegex = new Regex(@"(^|[^'])\{[0-9]+\}", RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

If the start of the line (`^`) _or_ anything that is _not_ a single quote (`[^']`) is followed by a replacement sequence (`\{[0-9]+\}`), that's a bad thing. So you test:

```
if (s_nonEnquotedDynamicContentRegex.IsMatch(text))
{
    // Report error.
}
```